src/parquet/column/reader.h

```diff
@@ -383,10 +383,10 @@ inline int64_t TypedColumnReader<DType>::Skip(int64_t num_rows_to_skip) {
 
       do {
         batch_size = std::min(batch_size, rows_to_skip);
-        values_read =
-            ReadBatch(batch_size, reinterpret_cast<int16_t*>(def_levels->mutable_data()),
-                reinterpret_cast<int16_t*>(rep_levels->mutable_data()),
-                reinterpret_cast<T*>(vals->mutable_data()), &values_read);
+        values_read = ReadBatch(static_cast<int>(batch_size),
```

Comment:
[PARQUET-1008](https://issues.apache.org/jira/browse/PARQUET-1008) created to address this